src/main/java/com/metamx/emitter/core/HttpPostEmitter.java

```diff
@@ -280,8 +283,11 @@ public void run()
           for (final List<byte[]> batch : batches) {
             log.debug("Sending batch to url[%s], batch.size[%,d]", url, batch.size());
 
+            byte[] decodedBatch = config.getCompress() ? serializeAndCompressBatch(batch) : serializeBatch(batch);
```

Comment:
suggest renaming it to something like "serializedBatch" or "payload", decoded in my understanding is the opposite when you got compressed or encoded input and produce raw bytes.